### PR TITLE
feat(turbo-ignore): pretty errors

### PR DESCRIPTION
**Pull request by tknickman**
Prettier errors for some common turbo-ignore exceptions

### CODE REVIEW

File: packages/turbo-ignore/__tests__/errors.test.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      err: `* reading yarn.lock: open /test/../yarn.lock: no such file or directory`,
+    });
+    expect(result.code).toBe("MISSING_LOCKFILE");
+    expect(result.level).toBe("warn");
+    expect(result.message).toBe(NON_FATAL_ERRORS.MISSING_LOCKFILE.message);
+  });
+
+  it("it detects errors when yarn lockfile is missing", async () => {
+    const result = shouldWarn({
+      err: `* reading pnpm-lock.yaml: open /test/../yarn.lock: no such file or directory`,
```
Comment (jaredpalmer) [inline packages/turbo-ignore/__tests__/errors.test.ts:24]: mismatch? should this say pnpm-lock.yaml?

File: packages/turbo-ignore/__tests__/errors.test.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    });
+    expect(result.code).toBe("MISSING_LOCKFILE");
+    expect(result.level).toBe("warn");
+    expect(result.message).toBe(NON_FATAL_ERRORS.MISSING_LOCKFILE.message);
+  });
+
+  it("it detects errors when npm lockfile is missing", async () => {
+    const result = shouldWarn({
+      err: `* reading package-lock.json: open /test/../yarn.lock: no such file or directory`,
```
Comment (jaredpalmer) [inline packages/turbo-ignore/__tests__/errors.test.ts:33]: same here?

File: packages/turbo-ignore/__tests__/ignore.test.ts [hunk 3]
```diff
@@ -40,7 +40,83 @@ describe("turboIgnore()", () => {
...
+
+    expect(mockExec).toHaveBeenCalledWith(
+      "npx turbo run build --filter=test-workspace...[HEAD^] --dry=json",
+      expect.anything(),
+      expect.anything()
+    );
+
+    validateLogs(
+      [
+        `turbo-ignore could not complete - not enough information available to compare`,
```
Comment (jaredpalmer) [inline packages/turbo-ignore/__tests__/ignore.test.ts:116]: to compare what to what? finish sentence

**Review by jaredpalmer — CHANGES_REQUESTED** [review]

File: packages/turbo-ignore/__tests__/errors.test.ts (hunk 1 above)
Comment (tknickman) [reply packages/turbo-ignore/__tests__/errors.test.ts:24]: 👍🏼 good catch

File: packages/turbo-ignore/__tests__/ignore.test.ts (hunk 3 above)
Comment (tknickman) [reply packages/turbo-ignore/__tests__/ignore.test.ts:116]: Updated to be more clear - good callout.

**Comment by tknickman** [comment]
@jaredpalmer updated

File: packages/turbo-ignore/src/types.ts [hunk 4]
```diff
@@ -1,3 +1,15 @@
+export type NonFatalErrorKeys =
```
Comment (gaspar09) [inline packages/turbo-ignore/src/types.ts:1]: This is a singular type, e.g. `NonFatalErrorKey`

**Review by gaspar09 — APPROVED** [review]

File: packages/turbo-ignore/src/types.ts (hunk 4 above)
Comment (tknickman) [reply packages/turbo-ignore/src/types.ts:1]: Good call, will fix

**Review by jaredpalmer — APPROVED** [review]

**Pull request merged**